### PR TITLE
[-] CORE: Removes unneeded I/O stat call

### PR DESCRIPTION
**Pull request by guestisp**

### CODE REVIEW

**Comment by guestisp** [comment]
AFAIK , `FrontController::display` calls `cccCss` in every page and `cccCss` always check for md5 and filemtime of each file. This could be a performance bottleneck and should be resolved.

File: classes/Media.php [hunk 1]
```diff
@@ -255,11 +255,11 @@ public static function getMediaPath($media_uri, $css_media_type = null)
 		// check if css files exists
 		if (!array_key_exists('host', $url_data))
 		{
-			if (!@filemtime($file_uri) || @filesize($file_uri) === 0)
+			if (!is_file($file_uri))
```
Comment (gRoussac) [inline classes/Media.php]: filesize is required here.

File: classes/Media.php [hunk 2]
```diff
@@ -255,11 +255,11 @@ public static function getMediaPath($media_uri, $css_media_type = null)
...
 			{
 				if (!defined('_PS_HOST_MODE_'))
 					return false;
-				elseif(!@filemtime($file_uri_host_mode) || @filesize($file_uri_host_mode) === 0)
+				elseif(!is_file($file_uri_host_mode))
```
Comment (gRoussac) [inline classes/Media.php]: filesize is required, we want to test the size. This is voluntary.

File: classes/Media.php (hunk 1 above)
Comment (guestisp) [inline classes/Media.php]: Why? You are just checking for true/false not for the real file size.

`@filesize($file_uri) === 0`

an empty file would be true, like with `is_file`

Comment (guestisp) [inline classes/Media.php]: You are right, my mistake. I'll fix.

**Comment by guestisp** [comment]
Casted `filesize` to `int` this returns 0 in case of empty file or missing file.
Now should be ok with just one stat call

**Comment by guestisp** [comment]
```
$ php -r "var_dump((int)@filesize('missingfile'));"
int(0)

$ php -r "var_dump((int)@filesize('emptyfile'));"
int(0)

$ php -r "var_dump((int)@filesize('fullfile'));"
int(31)
```

**Comment by jeromenadaud** [comment]
Hi,

Thank you for your PR, can you please sync your fork and branch in order to merge your PR.

Best regards.

**Comment by vTerenti** [comment]
(note: this is an automated message, but answering it will reach a real human )

Hi! Thank you for your pull request: your contribution to PrestaShop means a lot to us!

We are aware that this PR is still open after all this time, and we want to make sure that your hard work is not forgotten.

We are diving back into pull requests, and we want to work with you in order to get your PR to completion: we need your feedback!<br/>
Could you read [this article](http://build.prestashop.com/news/review-your-pull-requests/) from [the Build devblog](http://build.prestashop.com/), and take the time to review your pull requests, so that we can work with up-to-date information? You don't have to do it all right now, and every little step helps.

Again, thank you very much for this contribution, and for your renewed feedback on it!

**Comment by xBorderie** [comment]
Hey @guestisp ! Do you think you could port this change to the `develop` branch, so that it is possibly included in PrestaShop 1.7? Thank you!

**Comment by vTerenti** [comment]
Hello!

This PR has been automatically closed because the branch 1.6 does not exist anymore.

Best regards

**Pull request closed without merging**